commit 25d6eaf036718c408e3fca5c57c707b0c9ef44a8
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Tue Feb 19 15:06:31 2019 +0000

    gpu:vc4-fkms: Update driver to not use plane->crtc.
    
    Following on from
    commit 2f958af7fc248 ("drm/vc4: Stop updating plane->fb/crtc")
    do the same in the firmwarekms driver and look at plane_state->crtc
    instead.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/gpu/drm/vc4/vc4_firmware_kms.c b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
index e760b569c54d..4701f6850551 100644
--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -193,8 +193,8 @@ static void vc4_cursor_plane_atomic_update(struct drm_plane *plane,
 					   struct drm_plane_state *old_state)
 {
 	struct vc4_dev *vc4 = to_vc4_dev(plane->dev);
-	struct vc4_crtc *vc4_crtc = to_vc4_crtc(plane->crtc);
 	struct drm_plane_state *state = plane->state;
+	struct vc4_crtc *vc4_crtc = to_vc4_crtc(state->crtc);
 	struct drm_framebuffer *fb = state->fb;
 	struct drm_gem_cma_object *bo = drm_fb_cma_get_gem_obj(fb, 0);
 	dma_addr_t addr = bo->paddr + fb->offsets[0];
@@ -682,8 +682,6 @@ static int vc4_fkms_bind(struct device *dev, struct device *master, void *data)
 	drm_crtc_init_with_planes(drm, crtc, primary_plane, cursor_plane,
 				  &vc4_crtc_funcs, NULL);
 	drm_crtc_helper_add(crtc, &vc4_crtc_helper_funcs);
-	primary_plane->crtc = crtc;
-	cursor_plane->crtc = crtc;
 
 	vc4_encoder = devm_kzalloc(dev, sizeof(*vc4_encoder), GFP_KERNEL);
 	if (!vc4_encoder)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1355_25d6eaf036718c408e3fca5c57c707b0c9ef44a8.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1355_25d6eaf036718c408e3fca5c57c707b0c9ef44a8.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
